plugins/brain/__init__.py

```diff
@@ -1784,6 +1784,417 @@ def _get_target_view(ctx, target):
+        ),
+    )
+
+
+def find_near_duplicates(ctx):
+    target = ctx.params.get("target", None)
+    threshold = ctx.params.get("threshold", 0.3)
+    similarity_index = ctx.params.get("similarity_index", None)
+    roi_field = ctx.params.get("roi_field", None)
+    embeddings = ctx.params.get("embeddings", None) or None
+    model = ctx.params.get("model", None) or None
+    batch_size = ctx.params.get("batch_size", None)
+    num_workers = ctx.params.get("num_workers", None)
+    skip_failures = ctx.params.get("skip_failures", True)
+
+    # No multiprocessing allowed when running synchronously
+    if not ctx.delegated:
+        num_workers = 0
+
+    dataset = ctx.dataset
+    target_view = _get_target_view(ctx, target)
+
+    index = fob.compute_near_duplicates(
+        samples=target_view,
+        threshold=threshold,
+        similarity_index=similarity_index,
+        roi_field=roi_field,
+        embeddings=embeddings,
+        model=model,
+        batch_size=batch_size,
+        num_workers=num_workers,
+        skip_failures=skip_failures,
+    )
+
+    dups_view = index.duplicates_view()
+
+    repr_ids = list(index.neighbors_map.keys())
+    reps_view = dataset.select(repr_ids)
+
+    dataset.save_view("near duplicates", dups_view, overwrite=True)
+    dataset.save_view(
+        "representatives of near duplicates", reps_view, overwrite=True
+    )
+
+    return {
+        "num_total_dups": len(dups_view),
+        "num_unique_dups": len(repr_ids),
+    }
+
+
+class DeduplicateNearDuplicates(foo.Operator):
+    @property
+    def config(self):
+        return foo.OperatorConfig(
+            name="deduplicate_near_duplicates",
+            label="Deduplicate near duplicates",
+            description=(
+                "Delete all but one copy from each group of near duplicates"
+            ),
+            light_icon="/assets/icon-light.svg",
+            dark_icon="/assets/icon-dark.svg",
+            allow_delegated_execution=True,
+            allow_immediate_execution=True,
+            default_choice_to_delegated=True,
+            dynamic=True,
+        )
+
+    def resolve_input(self, ctx):
+        inputs = types.Object()
+
+        dataset = ctx.dataset
+        if dataset.has_saved_view("near duplicates"):
+            dups_view = dataset.load_saved_view("near duplicates")
+            reps_view = dataset.load_saved_view(
+                "representatives of near duplicates"
+            )
+
+            warning = types.Warning(
+                label=(
+                    f"Your last near duplicates scan detected {len(dups_view)} "
+                    f"near duplicates across {len(reps_view)} groupings. "
+                    "Executing this operator will delete all but one sample "
+                    "from each group."
+                )
+            )
+            inputs.view("warning", warning)
+        else:
+            find_near_duplicates_inputs(ctx, inputs)
+
+            warning = types.Warning(
```

Comment:
Added a warning for the case where there is no existing near duplicates scan explaining what will happen:

<img width="1154" height="717" alt="Screenshot 2025-07-24 at 2 16 10 AM" src="https://github.com/user-attachments/assets/99ddf138-6af2-429f-a051-2f8e6116f21a" />